blueprints/ember-cli-typescript/index.js

```diff
@@ -36,7 +37,23 @@ module.exports = {
     // Entity name is optional right now, creating this hook avoids an error.
   },
 
-  afterInstall() {
+  async afterInstall() {
+    if (existsSync('.gitignore')) {
+      await this.insertIntoFile('.gitignore', '\n# Ember CLI TypeScript\n.e-c-ts');
+    }
+
+    if (existsSync('.vscode/settings.json')) {
+      // NOTE: this may or may not actually work -- it assumes that
+      // `"tmp:": true` is in fact set.
+      await this.insertIntoFile('.vscode/settings.json', '",\n.e-c-ts": true', {
+        after: '"tmp": true',
+      });
+    }
+
+    // TODO: same for Atom
+    // TODO: same for Sublime
+    // TODO: same for IntelliJ
```

Comment:
Would this whole `afterInstall` hook be avoidable if the output directory lived in either system temp (`os.tmpdir()`) or piggybacked off the local broccoli temporary directory (which is presumably already set up to be ignored by the user's editor, git, npm, etc)?

Reply:
I would think so. And if we can piggy-back off of `os.tmpdir()`, that seems reasonable enough to me, though making sure *that* gets pulled into the Broccoli tree seems… iffy?

Reply:
We rely on watchman (or other watcher, but I've only tested with watchman) to pick up the compiled js.

Watchman is set up to monitor the root directory of the app, minus the excluded directories in the default `.watchmanconfig`:
```json
{
  "ignore_dirs": ["tmp", "dist"]
}

```

So we can't put the files in `tmp` or some other system directory

Reply:
It's possible we could add another watched directory somehow but I haven't explored the APIs for it

Reply:
My understanding is it would end up adding that directory as a watchman root, but I'm pretty unfamiliar with the details there, so grain of salt. Best bet is probably just to try it and see what happens ¯\\\_(ツ)_/¯